TESTING.md

```diff
@@ -0,0 +1,69 @@
+  - Call `LocalGcdHelper.start(<project ID>, <port number>)` before running your tests.  Save the `LocalGcdHelper` object returned so that you can stop the emulator later.
+
+2. In your program, create and use a datastore whose host is set host to `localhost:<port number>`.  For example, 
+  ```java
+  DatastoreOptions options = DatastoreOptions.builder()
+      .projectId(PROJECT_ID)
+      .host("localhost:8080")
+      .build();
+  Datastore localDatastore = DatastoreFactory.instance().get(options);
+  ```
+3. Run your tests.
+
+4. Stop the local datastore emulator.
+  - If you ran `LocalGcdHelper.java`'s `main` function to start the emulator, run `LocalGcdHelper.java`'s `main` method with arguments `STOP` and (optionally) `--port=<port number>`.  If the port is not supplied, the program will attempt to close the last port started.
+  - If you ran `LocalGcdHelper.start()` to start the emulator, call the `stop()` method on the `LocalGcdHelper` object returned by `LocalGcdHelper.start()`.
+
+#### On a remote machine
+
+You can test against a remote datastore emulator as well.  To do this, set the `DatastoreOptions` project endpoint to the hostname of the remote machine, like the example below.
+
+  ```java
+  DatastoreOptions options = DatastoreOptions.builder()
+      .projectId(PROJECT_ID)
+      .host("http://<hostname of machine>")
+      .build();
+  Datastore localDatastore = DatastoreFactory.instance().get(options);
+  ```
+
+Note that the remote datastore must be running before your tests are run.  Also note that the `host` argument must start with "http://" or "https://" if you are testing with a remote machine.
+
+
+### Testing interactions with Storage
+
+There currently isn't an emulator for Google Cloud Storage, so an alternative is to create a test project.  `RemoteGcsHelper` contains convenience methods to make setting up and cleaning up the test project easier.  To use this class, follow the steps below:
+
+1. Create a test Google Cloud project.
+
+2. Create and download a JSON service account credentials file from the Google Developer's Console.  See more about this on the [Google Cloud Platform Storage Authentication page][cloud-platform-storage-authentication]. 
+
+3. Set environment variables `GCLOUD_TESTS_PROJECT_ID` and `GCLOUD_TESTS_KEY` according to your test project's ID and the location of the newly-downloaded JSON key file.  On linux and mac, for example,
+  ```
+  export GCLOUD_TESTS_PROJECT_ID=<project id>
+  export GCLOUD_TESTS_KEY=/path/to/JSON/key.json
+  ```
+
+4. Create and and use a `RemoteGcsHelper` object.
+Here is an example that uses the `RemoteGcsHelper` to create a bucket and clear the bucket at the end of the test.
+```java
+RemoteGcsHelper gcsHelper = RemoteGcsHelper.create();
```

Comment:
@mziccard I wonder if we should also supply an option to create a `RemoteGcsHelper` with a given projectId and a key reference (that will work without the environment variables or override them if exists)?

I think environment variables work well for integration tests but not so for regular unit-tests.